src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php

```diff
@@ -114,35 +112,47 @@ public function getData($id)
     public function getDefaultData()
     {
         return $this->addShortcutData([
-            'basic' => [
+            'header' => [
                 'type' => ProductType::TYPE_STANDARD,
             ],
-            'manufacturer' => [
-                'manufacturer_id' => NoManufacturerId::NO_MANUFACTURER_ID,
+            'basic' => [
+                'manufacturer' => NoManufacturerId::NO_MANUFACTURER_ID,
             ],
             'stock' => [
-                'quantity' => 0,
-                'minimal_quantity' => 0,
+                'quantities' => [
+                    'quantity' => 0,
+                    'minimal_quantity' => 0,
+                ],
             ],
-            'price' => [
-                'price_tax_excluded' => 0,
-                'price_tax_included' => 0,
+            'pricing' => [
+                'retail_price' => [
+                    'price_tax_excluded' => 0,
+                    'price_tax_included' => 0,
+                ],
                 'tax_rules_group_id' => $this->mostUsedTaxRulesGroupId,
                 'wholesale_price' => 0,
-                'unit_price' => 0,
+                'unit_price' => [
+                    'price' => 0,
+                ],
             ],
             'shipping' => [
-                'width' => 0,
-                'height' => 0,
-                'depth' => 0,
-                'weight' => 0,
+                'dimensions' => [
+                    'width' => 0,
+                    'height' => 0,
+                    'depth' => 0,
```

Comment:
When I see such big arrays I wonder if we should create DTO for them.

The DTO would have 2 responsibilities
1. Contain the structure of the data => immutable and a bit more performing
2. It would have a `public function toFormData()` function that would output the array structure we have here which would allow us to do

 ```
 $myDto  = new ShortcutDataDTO(...);
 
 return $this->addShortcutData(myDto->toFormData());
 ```

Reply:
(suggestion, not for the PR)

Reply:
That we be a doublon, the purpose of this service is exactly to do what you suggest ^^

It gets the result from the CQRS queries (which are already DTOs) and format them into an array format that is adapted to the form structure

So adding yet another layer of DTO would be a doublon with the QueryResult ones